src/compas_fab/robots/robot.py

```diff
@@ -864,15 +873,40 @@ def forward_kinematics(self, configuration, group=None):
         response = self.client.forward_kinematics(
             joint_positions, base_link, group, joint_names, ee_link)
 
-        # TODO implement response types
-
         frame_RCF = response.pose_stamped[0].pose.frame
         frame_RCF.point *= self.scale_factor
         response.frame_RCF = frame_RCF
         response.frame_WCF = self.represent_frame_in_WCF(frame_RCF, group)
 
         return response
 
+    def forward_kinematics_robot_model(self, configuration, group=None, link_name=None):
```

Comment:
This name makes me feel something is not right. It's too long, and redundant in the sense that there is a `self.model` but still we need to say `forward_kinematics_robot_model`.

One option would be to change the `forward_kinematics` method, adding an optional parameter and depending on its value, use the `self.model` or use `self.client.forward_kinematics` method.

The name and possible values of this parameter probably need some thought or discussion, but one option would be to call it `backend`. Then the cases to handle are:
* `backend=None and self.client != None` => use client service
* `backend=None and self.client = None` => use local robot model resolution
* `backend='local'` => use local robot model resolution
* `backend != None and backend != 'local'` => use client service and pass that value down to the `self.client.forward_kinematics` method, because the client might do some further selection (e.g. KDL vs UR kinematic model, etc)

If this proves to look like a nice API, we should extend it to the IK side.